### ISSUE TITLE
Can't serve jsx-runtime as JavaScript on deno.land/x

### ISSUE THREAD

**Issue by connorlogin** [connorlogin/deno-preact#2]
Deno's registry can't yet serve arbitrary files with the x-typescript-types header, and any file without an extension will be served as octet-stream.

This is a blocking probem. `@jsxImportSource` is the [docs-recommended ](https://deno.land/manual/jsx_dom/jsx) way to use JSX with Deno, and the pragma insinuates a trailing `/jsx-runtime` when resolving the path given. However, if the path given isn't interpreted as a module either by the HTTP headers or by the file extension, Deno will fail to load it because it doesn't recognize it as a module. [Here's where the error message comes from in `deno_graph`](https://github.com/denoland/deno_graph/blob/20ae69f4eb3646e48845ec09922e2df792261d01/src/graph.rs#L202). i.e. `@jsxImportSource` doesn't work with any on-disk `jsx-runtime` file, and it also doesn't work with deno.land/x. (As far as I can tell.)

Without the ability to serve proper JS/TS headers, Preact on deno.land/x just isn't as useful as esm.sh.

I'm going to put a note in the readme pointing to https://esm.sh/preact, and another note for Preact maintainers if they'd like to take control of this repo. (I should've done more testing/research before registering it.)

**Comment by developit** [comment]
Hiya! Preact maintainer here - think it might be possible to coordinate releasing Preact officially for Deno?

**Comment by connorlogin** [comment]
Yes! Super glad you asked, `deno.land/x/preact` rightfully belongs to you and it doesn't feel right that I have it lol. I've transferred it to you on GitHub, but I had to change the name to "preact-deno" because "developit/deno-preact" already exists. [Control of this repo equates to control of `deno.land/x/preact`](https://github.com/denoland/deno_registry2/issues/24), so the name shouldn't matter.

Sorry for the junk 0.0.x releases, I didn't really know what I was doing. I tried to just build Preact and manually package up the types when I was attempting to get it off the ground. Some things worked but not everything, notably the problem described in this github issue. As far as I know `@jsxImportSource` is the still the only drawback. I'm pretty sure using the `@jsx` pragma in .tsx files should get jsx syntax working, but I don't remember if I tried that or not when I was working on this.

I'm excited to see this repo move to your username, can't wait to have Preact on deno.land.

**Comment by connorlogin** [comment]
Hey @developit. I deleted my last response to your comment because in hindsight I was a little too quick to pass off ownership/responsibility of this repo without first checking with you on how you'd like to collaborate.

My plan with this was to finish v0.1 of https://github.com/connorlogin/cav and then work on pinning the latest version of Preact to https://deno.land/x/preact via automation. I want to write some docs on how to use Preact with Cav, and I don't want to use https://esm.sh/preact when writing them because it's not an official Deno CDN and I'm aiming for a particular aesthetic. My timeline for cav@0.1.0 is within the next few weeks.

I'm happy to continue with that plan if that's okay by you, or if you'd like to do something different I've created #3 for tracking progress and planning. Re: the transfer request I initiated previously, I'm up for maintaining ownership, also up for handing it over if you'd like more control. Either way is fine by me.
